server/src/pages/shotindex/view.js

```diff
@@ -34,7 +34,7 @@ class Body extends React.Component {
         <div className="column-space full-height default-color-scheme">
           <div id="shot-index-header" className="header">
             <h1><a href="/shots">Firefox <strong>Screenshots</strong> <sup>Beta</sup></a></h1>
-            <form onSubmit={ this.onSubmitForm.bind(this) }>
+            <form onSubmit={ this.onSubmitForm.bind(this) } style={{display: this.state.displaySearch}}>
```

Comment:
There's no need for state here, the props can be used alone. Also I'd rather the form just be left out, instead of rendered but hidden.

So if you move the search form to a method, it would just be like `{ this.props.disableSearch ? null : this.renderSearchForm() }`